src/api/apiRequest.ts

```diff
@@ -0,0 +1,55 @@
+const BASE_URL = import.meta.env.VITE_API_BASE_URL;
+
+interface ApiOptions {
+  method?: string;
+  body?: unknown;
+  headers?: Record<string, string>;
+}
+
+export const apiRequest = async <T>(
+  endpoint: string,
+  options: ApiOptions = {}
+): Promise<T> => {
+  const url = `${BASE_URL}${endpoint}`;
+  const headers = {
+    ...options.headers,
+  };
+
+  try {
+    const response = await fetch(url, {
+      method: options.method || 'GET',
+      headers,
+      body: options.body ? JSON.stringify(options.body) : undefined,
+    });
+
+    if (!response.ok) {
+      let errorMessage: string;
+
+      if (response.status === 401) {
+        errorMessage = '인증에 실패했습니다. 다시 시도해주세요.';
+      } else if (response.status === 404) {
+        errorMessage = '요청한 페이지(데이터)를 찾을 수 없습니다.';
+      } else if (response.status === 500) {
+        errorMessage = '서버에 오류가 발생했습니다. 잠시 후 다시 시도해주세요.';
+      } else {
+        errorMessage = response.statusText;
+      }
+
+      throw new Error(errorMessage);
+    }
```

Comment:
```suggestion
    validateApiResponse(response)
```

이렇게 검증 함수를 하나 만들어서 관리해주면 어떨까 싶어요!
이러면 검증에 대한 테스트도 더 수월해진답니다 ㅎㅎ